code/game/machinery/autolathe.dm

```diff
@@ -103,49 +103,111 @@
 
 
 /obj/machinery/autolathe/attackby(var/obj/item/I, var/mob/user)
+	if(default_deconstruction(I, user))
+		return
+
+	if(default_part_replacement(I, user))
+		return
+
+	user.set_machine(src)
+	ui_interact(user)
+
+
+/obj/machinery/autolathe/attack_hand(mob/user as mob)
+	if(..())
+		return
+
+	user.set_machine(src)
+	ui_interact(user)
+
+/obj/machinery/autolathe/Topic(href, href_list)
 
-	if(istype(I,/obj/item/weapon/disk/autolathe_disk))
+	if(..())
+		return
+
+	add_fingerprint(usr)
+
+	usr.set_machine(src)
+
+	if(href_list["eject_disk"] && disk)
+		disk.forceMove(src.loc)
+
+		if(isliving(usr))
+			var/mob/living/L = usr
+			if(istype(L))
+				L.put_in_active_hand(disk)
+
+		disk = null
+
+	if(href_list["insert"])
+		eat(usr)
+
+	if(!making_recipe)
+		if(href_list["eject_material"])
+			var/material = href_list["eject_material"]
+			var/material/M = get_material_by_name(material)
+
+			if(!M.stack_type)
+				return
+
+			var/obj/item/stack/material/sheetType = M.stack_type
+			var/perUnit = initial(sheetType.perunit)
+
+			var/num = input("Enter sheets count to eject. 0-[round(stored_material[material]/perUnit)]","Eject",0) as num
+			num = min(max(num,0), round(stored_material[material]/perUnit))
+
+			eject(material, num)
+
+		if(href_list["print_one"] && disk)
+			if(!making_recipe)
+				make(text2path(href_list["print_one"]),1)
+
+		if(href_list["print_several"] && disk)
+			if(!making_recipe)
+				var/num = input("Enter items count to print.","Print") as num
+				make(text2path(href_list["print_several"]),num)
+
+	if(href_list["abort_print"])
+		making_recipe = null
+		making_left = 0
+		making_total = 0
+		disk_error = FALSE
+		not_enough_resources = FALSE
+
+	nanomanager.update_uis(src)
+
+
+/obj/machinery/autolathe/proc/eat(var/mob/living/user)
+	if(!istype(user))
+		return
+
+	var/obj/item/eating = user.get_active_hand()
+
+	if(!istype(eating))
+		return
+
+	if(istype(eating,/obj/item/weapon/disk/autolathe_disk))
 		if(disk)
 			user << SPAN_NOTICE("There's already a disk inside the autolathe.")
 			return
 		user.drop_item()
```

Comment:
Better use 
`user.unEquip(item, new location)` 
instead
```
user.drop_item()
item.forceMove(new location)
```
user.drop_item() - should be used only for situation where user must drop any item (no matter what it was) from hands.